test/unit/org/apache/cassandra/cql3/validation/entities/FrozenCollectionsTest.java

```diff
@@ -258,6 +258,84 @@ public void testClusteringKeyUsage() throws Throwable
         );
     }
 
+    @Test
+    public void testClusteringKeyUsageWithReverseOrder() throws Throwable
```

Comment:
@adelapena given the codes fixes touch Lists, Sets and Maps I think we should test the 3 and not Sets alone. Wdyt?

Reply:
We have specific tests in `testLiteralReversedList`, `testLiteralReversedSet` and `testLiteralReversedMap`. This test and `testNestedClusteringKeyUsageWithReverseOrder` are counterparts for the existing `testClusteringKeyUsage` and `testNestedClusteringKeyUsage`, which also use only a single set type. Do you think we should also multiplex those, and have:
* testClusteringKeyUsageSet
* testClusteringKeyUsageList
* testClusteringKeyUsageMap
* testClusteringKeyUsageWithReverseOrderSet
* testClusteringKeyUsageWithReverseOrderList
* testClusteringKeyUsageWithReverseOrderMap
* testNestedClusteringKeyUsageSet
* testNestedClusteringKeyUsageList
* testNestedClusteringKeyUsageMap
* testNestedClusteringKeyUsageWithReverseOrderSet
* testNestedClusteringKeyUsageWithReverseOrderList
* testNestedClusteringKeyUsageWithReverseOrderMap

Reply:
Sorry, `testNestedClusteringKeyUsage` and `testNestedClusteringKeyUsageWithReverseOrder` already include the three collection types, so it would be:
* testClusteringKeyUsageSet
* testClusteringKeyUsageList
* testClusteringKeyUsageMap
* testClusteringKeyUsageWithReverseOrderSet
* testClusteringKeyUsageWithReverseOrderList
* testClusteringKeyUsageWithReverseOrderMap
* testNestedClusteringKeyUsageSet
* testNestedClusteringKeyUsageWithReverseOrder

Reply:
@adelapena that's what I would do. Given List and Set tests are identical despite some minor syntax diffs I'll presume they can be easily folded into a common main test method + a Map one. But I am OCD :shrug: so I leave it at your discretion.

Reply:
Oh, I pushed a commit to the 2.2 PR with the changes before reading this. It folds the three collection types, actually, and also includes `testPartitionKeyUsage`.